### PR TITLE
BAEL-4558: Article for stopping execution after a certain time

### PR DESCRIPTION
**Pull request by sahilsingla112**

### CODE REVIEW

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+    public static void main(String[] args) {
+        StopExecution stopExecution = new StopExecution();
+        //stopExecution.testUsingLoop();
+        //stopExecution.testTimer();
+        stopExecution.testScheduledExecutor();
+        LOG.info("done");
+    }
+
+
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:32]: remove the extra blank line

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    public void testUsingLoop(){
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:33]:
```suggestion
    public void testUsingLoop() {
```

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        long start = System.currentTimeMillis();
+        long end = start + 5000;
+        List<String> items = new ArrayList<>();
+        int counter = 0;
+
+        // Let this loop run only upto 5 seconds
+        while (System.currentTimeMillis() < end && counter < items.size())
+        {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:41]: opening brace should be at the end of a line, not on a line by itself

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+        while (thread.getState() != Thread.State.TERMINATED){
+            LOG.info(thread.getState().name());
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+    public static void testExecutor(){
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:77]: need a blank line between methods

Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:77]:
```suggestion
    public static void testExecutor() {
```

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+            LOG.error("Calculation took to long");
+        } catch (final Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            service.shutdown();
+        }
+    }
+
+
+    public void testExecutor2(){
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:106]:
```suggestion
    public void testExecutor2() {
```

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+            f.cancel(true);
+            LOG.error("Calculation took to long");
+        } catch (final Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            service.shutdown();
+        }
+    }
+
+
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:105]: reduce to one blank line

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+            f.cancel(true);
+            LOG.error("Calculation took to long");
+        } catch (final Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            service.shutdownNow();
+        }
+    }
+
+    public void testScheduledExecutor(){
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:123]:
```suggestion
    public void testScheduledExecutor() {
```

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+        ScheduledExecutorService executor = Executors.newScheduledThreadPool(2);
+        Future future = executor.submit(new LongRunningTask());
+        executor.schedule(new Runnable(){
+            public void run(){
+                future.cancel(true);
+            }
+        }, 1000, TimeUnit.MILLISECONDS);
+        executor.shutdown();
+    }
+    public void testThreadAndInterrupt(){
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:134]:
```suggestion
    public void testThreadAndInterrupt() {
```

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+            t.start();
+            while (t.isAlive() && System.currentTimeMillis() < end){
+                Thread.sleep(50);
+            }
+            t.interrupt();
+        } catch (final Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+    public void testTimer(){
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:151]:
```suggestion
    public void testTimer() {
```

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+                Thread.sleep(10000);
+            }
+            catch (InterruptedException ie)
+            {
+                LOG.info("MyRunnable interrupted...");
+            }
+        }
+    }
+
+
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:174]: one blank line

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 11]
```diff
@@ -0,0 +1,268 @@
...
+    class TimeOutTask extends TimerTask {
+        private Thread t;
+        private Timer timer;
+
+        TimeOutTask(Thread t, Timer timer){
+            this.t = t;
+            this.timer = timer;
+        }
+        public void run() {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:183]: separate methods with a blank line

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 12]
```diff
@@ -0,0 +1,268 @@
...
+            if (t != null && t.isAlive()) {
+                t.interrupt();
+                timer.cancel();
+            }
+        }
+    }
+
+    class LongRunningTask implements Runnable{
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:191]:
```suggestion
    class LongRunningTask implements Runnable {
```

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 13]
```diff
@@ -0,0 +1,268 @@
...
+        @Override
+        public void run() {
+            longRunningSort();
+        }
+
+        private void longRunningOperation(){
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:197]:
```suggestion
        private void longRunningOperation() {
```

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 14]
```diff
@@ -0,0 +1,268 @@
...
+            LOG.info("long Running operation started");
+
+            try {
+                //Thread.sleep(500);
+                longFileRead();
+                LOG.info("long running operation finished");
+            }
+            catch (InterruptedException e){
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:205]:
```suggestion
            catch (InterruptedException e) {
```

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 15]
```diff
@@ -0,0 +1,268 @@
...
+                LOG.info("long Running operation interrupted");
+            }
+        }
+
+        private void longRunningSort(){
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:210]:
```suggestion
        private void longRunningSort() {
```

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 16]
```diff
@@ -0,0 +1,268 @@
...
+                    if (minIndex != i) {
+                        int temp = numbers.get(i);
+                        numbers.set(i, numbers.get(minIndex));
+                        numbers.set(minIndex, temp);
+                    }
+                    throwExceptionOnThreadInterrupt();
+                }
+                LOG.info("Index position: " + i);
+                LOG.info("Long running task finished");
+            }catch (InterruptedException e){
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:237]:
```suggestion
            }catch (InterruptedException e) {
```

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 17]
```diff
@@ -0,0 +1,268 @@
...
+                LOG.info("long Running operation interrupted");
+            }
+        }
+
+        private void longFileRead() throws InterruptedException{
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:242]:
```suggestion
        private void longFileRead() throws InterruptedException {
```

File: core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java [hunk 18]
```diff
@@ -0,0 +1,268 @@
...
+                while (data != -1) {
+                    char theChar = (char) data;
+                    data = inputStreamReader.read();
+                    throwExceptionOnThreadInterrupt();
+                }
+            } catch (IOException e){
+                LOG.error("Exception: ", e);
+            }
+        }
+        private void throwExceptionOnThreadInterrupt() throws InterruptedException{
```
Comment (KevinGilmore) [inline core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/stopexecution/StopExecution.java:259]:
```suggestion
        private void throwExceptionOnThreadInterrupt() throws InterruptedException {
```

**Review by KevinGilmore — COMMENTED** [review]
Formatting/style issues, mostly with regard to spacing

**Review by sahilsingla112 — COMMENTED** [review]
Applied the formmatter.

**Pull request merged**